In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

In [2]:
products = pd.read_csv("Modificados - Atmira_Pharma_Visualization/products.csv")

In [6]:
products

product_id           sku  \
0            7349        165774   
1            7347        165776   
2           50282        230154   
3            7348        165775   
4           24517        011905   
...           ...           ...   
27948       22432   433363.AZUL   
27949       30090  10.22.12.256   
27950       97587      PURO0113   
27951        3693        211391   
27952       51856        PT3603   

                                                    name    marca_value  \
0         Pañales CHELINO LOVE Talla3 4-10 KG 36unidades        Chelino   
1        Pañales CHELINO LOVE Talla5 13-18 KG 30unidades        Chelino   
2                           Pañales Pingo Maxi T4 40 uds          Pingo   
3         Pañales CHELINO LOVE Talla4 9-15 KG 34unidades        Chelino   
4                           Dodot Pañal T4 9-15Kg 30unds          Dodot   
...                                                  ...            ...   
27948      Sujetachupetes TOMMEE TIPPEE (2 tons de azul)  Tommee tippee   
27949  Crème de profunda esfoliante Apivita Rose pimenta        Apivita   
27950  Corrector Líquido Sublime Ecológico PuroBio To...        PuroBio   
27951         Dermatix Gel Silicone Cicatrizes 15 gramas       Dermatix   
27952                 Plantoys Cocina Infantil de Madera       Plantoys   

                                       short_description    analytic_category  \
0      <p>Los pañales <strong>Chelino</strong>, que p...             Infantil   
1      <p>Los pañales Chelino que presentan un novedo...             Infantil   
2      <p>Pingo es el primer pañal ecológico con cuat...             Infantil   
3      <p>Los pañales Chelino, que presentan un noved...             Infantil   
4      <p>¿Qué le pides a un pañal? ¿Absorción máxima...                  NaN   
...                                                  ...                  ...   
27948  <p>Prácticos sujetachupetes infantiles ideales...             Infantil   
27949  <p>Apivita Rose Pepper Crema Exfoliante Profun...              Higiene   
27950  <p>Corrector líquido apto para veganos. Muy cu...  Cosmética y Belleza   
27951  <p>Gel que elimina o reduce las cicatrices. Es...  Cosmética y Belleza   
27952  <p>Preciosa cocina elaborada en madera maciza,...             Infantil   

                                                 picture  
0      https://www.mifarma.es/media/catalog/product/1...  
1      https://www.mifarma.es/media/catalog/product/1...  
2      https://www.mifarma.es/media/catalog/product/2...  
3      https://www.mifarma.es/media/catalog/product/1...  
4      https://www.mifarma.es/media/catalog/product/0...  
...                                                  ...  
27948  https://www.mifarma.es/media/catalog/product/4...  
27949  https://www.mifarma.es/media/catalog/product/1...  
27950  https://www.mifarma.es/media/catalog/product/p...  
27951  https://www.mifarma.es/media/catalog/product/2...  
27952  https://www.mifarma.es/media/catalog/product/p...  

[27953 rows x 7 columns]

In [4]:
products.shape

(27953, 7)

In [5]:
products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27953 entries, 0 to 27952
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   product_id         27953 non-null  int64 
 1   sku                27953 non-null  object
 2   name               27939 non-null  object
 3   marca_value        27921 non-null  object
 4   short_description  27930 non-null  object
 5   analytic_category  24416 non-null  object
 6   picture            27947 non-null  object
dtypes: int64(1), object(6)
memory usage: 1.5+ MB


In [33]:
products[products.duplicated()].shape

(0, 7)

### id y sku

In [28]:
products['product_id'].value_counts()

23299    4
2493     4
13134    4
2491     4
5573     4
        ..
21041    1
24604    1
52213    1
70477    1
51856    1
Name: product_id, Length: 23183, dtype: int64

In [63]:
products[products.duplicated(["product_id"])]

product_id           sku  \
3067         5572        358838   
3068         4629        358820   
3097         5574        150402   
3098         5573        357905   
3269         5575        185501   
...           ...           ...   
27946       46420      42401068   
27947       46414      42401568   
27948       22432   433363.AZUL   
27949       30090  10.22.12.256   
27951        3693        211391   

                                                    name    marca_value  \
3067      Aceite Corporal de Calendula Bebe Weleda 200ml         Weleda   
3068             Crema Facial Bebe Calendula Weleda 50ml         Weleda   
3097   Champu y Gel de Ducha de Calendula Bebe Weleda...         Weleda   
3098       Leche Corporal de Calendula Bebe Weleda 200ml         Weleda   
3269               Baño de Crema de Calendula Bebe 200ml         Weleda   
...                                                  ...            ...   
27946  Unidades de médio fluxo 2 Tommee Tippee tetina...  Tommee tippee   
27947  Tetina de fluxo Tommee Tippee n º 3 unidades d...  Tommee tippee   
27948      Sujetachupetes TOMMEE TIPPEE (2 tons de azul)  Tommee tippee   
27949  Crème de profunda esfoliante Apivita Rose pimenta        Apivita   
27951         Dermatix Gel Silicone Cicatrizes 15 gramas       Dermatix   

                                       short_description    analytic_category  \
3067   <p>Tratamiento protector para pieles sensibles...             Infantil   
3068   <p>Crema facial para bebés rica en extracto de...             Infantil   
3097   <p>Limpieza extra suave para cabello y piel de...             Infantil   
3098   <p>Leche corporal indicada para el cuidado dia...             Infantil   
3269   <p>Crema de caléndula ideal para los primeros ...             Infantil   
...                                                  ...                  ...   
27946  <p>La tetina Ultra de Tommee Tippee está diseñ...             Infantil   
27947  <p>La tetina Ultra de Tommee Tippee está diseñ...             Infantil   
27948  <p>Prácticos sujetachupetes infantiles ideales...             Infantil   
27949  <p>Apivita Rose Pepper Crema Exfoliante Profun...              Higiene   
27951  <p>Gel que elimina o reduce las cicatrices. Es...  Cosmética y Belleza   

                                                 picture  
3067   https://www.mifarma.es/media/catalog/product/3...  
3068   https://www.mifarma.es/media/catalog/product/3...  
3097   https://www.mifarma.es/media/catalog/product/1...  
3098   https://www.mifarma.es/media/catalog/product/3...  
3269   https://www.mifarma.es/media/catalog/product/1...  
...                                                  ...  
27946  https://www.mifarma.es/media/catalog/product/4...  
27947  https://www.mifarma.es/media/catalog/product/4...  
27948  https://www.mifarma.es/media/catalog/product/4...  
27949  https://www.mifarma.es/media/catalog/product/1...  
27951  https://www.mifarma.es/media/catalog/product/2...  

[4770 rows x 7 columns]

In [25]:
sum(products["product_id"].value_counts() > 1)

4334

In [26]:
sum(products["product_id"].value_counts() <= 1)

18849

In [43]:
sum(products["sku"].value_counts() <= 1)

18902

In [44]:
sum(products["sku"].value_counts() > 1)

4314

In [51]:
sum(products["sku"].value_counts() > 1) & sum(products["product_id"].value_counts() > 1)

4298

In [29]:
products[products["product_id"] == 2493]

product_id     sku                                           name  \
2191         2493  152792            Crema Reafirmante Woman Isdin 150ml   
7400         2493  152792          Velastina reafirmante pós-parto 150ml   
23940        2493  152792        Woman Isdin reafirmante pós-parto 150ml   
27508        2493  152792  Creme Isdin Woman reafirmante pós-parto 150ml   

      marca_value                                  short_description  \
2191        Isdin  <p>Esta crema está especialmente indicada para...   
7400        Isdin  <p>Esta crema está especialmente indicada para...   
23940       Isdin  <p>Esta crema está especialmente indicada para...   
27508       Isdin  <p>Esta crema está especialmente indicada para...   

      analytic_category                                            picture  
2191           Infantil  https://www.mifarma.es/media/catalog/product/2...  
7400           Infantil  https://www.mifarma.es/media/catalog/product/2...  
23940          Infantil  https://www.mifarma.es/media/catalog/product/2...  
27508          Infantil  https://www.mifarma.es/media/catalog/product/2...

### pictures

In [67]:
products["picture"].isna().sum()

6

In [68]:
products["picture"].value_counts()

https://www.mifarma.es/media/catalog/product/1/5/159387_234444_.jpg     4
https://www.mifarma.es/media/catalog/product/1/9/197950_.jpg            4
https://www.mifarma.es/media/catalog/product/1/5/159078_.jpg            4
https://www.mifarma.es/media/catalog/product/1/8/183286_.jpg            4
https://www.mifarma.es/media/catalog/product/4/4/448478417040_.jpg      4
                                                                       ..
https://www.mifarma.es/media/catalog/product/8/0/802569.naranja_.jpg    1
https://www.mifarma.es/media/catalog/product/1/8/180896_.jpg            1
https://www.mifarma.es/media/catalog/product/b/c/bc875_.jpg             1
https://www.mifarma.es/media/catalog/product/1/5/157168_.jpg            1
https://www.mifarma.es/media/catalog/product/p/t/pt3603_.jpg            1
Name: picture, Length: 23175, dtype: int64

In [36]:
products["picture"].tail()

27948    https://www.mifarma.es/media/catalog/product/4...
27949    https://www.mifarma.es/media/catalog/product/1...
27950    https://www.mifarma.es/media/catalog/product/p...
27951    https://www.mifarma.es/media/catalog/product/2...
27952    https://www.mifarma.es/media/catalog/product/p...
Name: picture, dtype: object

### names

In [42]:
products[products["name"].isna()]

product_id     sku name marca_value  \
18323         380  170696  NaN       Vichy   
18324         647  203465  NaN       Avene   
18326        3717  369702  NaN   Martiderm   
18327        3721  154563  NaN   Martiderm   
18331         645  356709  NaN       Avene   
18332        1797  255620  NaN       Isdin   
18334        3501  238436  NaN     Klorane   
18337        3861  234790  NaN    Sesderma   
18339        2671  194340  NaN      Uriage   
18340        1730  159316  NaN       Isdin   
18341        3947  160260  NaN   Neostrata   
18350        1336  330669  NaN   Bepanthol   
18351        2354  060095  NaN        Evax   
18352        2524  310601  NaN       Goibi   

                                       short_description    analytic_category  \
18323  <p>Desodorante antitranspirante sin alcohol co...              Higiene   
18324  <p>El Agua termal de Avène es un verdadero pri...  Cosmética y Belleza   
18326  <p>Tratamiento despigmentante intensivo, con f...  Cosmética y Belleza   
18327  <p>Stick con alta protección solar SPF50+ y ef...  Cosmética y Belleza   
18331  <p>El <strong>agua termal de Avene</strong> es...  Cosmética y Belleza   
18332  <p>Champú de uso diario que deja el cabello li...              Higiene   
18334  <p>Proporciona al cabello vigor y brillo, neut...              Higiene   
18337  <p>Esta crema exfoliente contiene cristales de...              Higiene   
18339  <p>Fluido Dermo indicado para limpiar las impu...              Higiene   
18340  <p>Protege del sol e hidrata la frágil y delic...  Cosmética y Belleza   
18341  <p>Fluido <strong>oil free</strong> indicado p...  Cosmética y Belleza   
18350  <p>Pomada protectora para el cuidado de las ir...             Infantil   
18351  <p>Compresa normal Cottonlike de Evax indicada...                  NaN   
18352  <p>Spray Antimosquitos especial para viajes tr...              Higiene   

                                                 picture  
18323  https://www.mifarma.es/media/catalog/product/4...  
18324  https://www.mifarma.es/media/catalog/product/2...  
18326  https://www.mifarma.es/media/catalog/product/3...  
18327  https://www.mifarma.es/media/catalog/product/1...  
18331  https://www.mifarma.es/media/catalog/product/4...  
18332  https://www.mifarma.es/media/catalog/product/2...  
18334  https://www.mifarma.es/media/catalog/product/2...  
18337  https://www.mifarma.es/media/catalog/product/2...  
18339  https://www.mifarma.es/media/catalog/product/1...  
18340  https://www.mifarma.es/media/catalog/product/1...  
18341  https://www.mifarma.es/media/catalog/product/1...  
18350  https://www.mifarma.es/media/catalog/product/4...  
18351  https://www.mifarma.es/media/catalog/product/0...  
18352  https://www.mifarma.es/media/catalog/product/3...

In [64]:
products["name"].value_counts().head(10)

Avene Couvrance Compacto                                 4
Medela Embudo Personal Fit                               3
IAP Pharma Perfume mulher n º 150 ml                     3
Medela Top de Extracción Fácil Sujetador Manos Libres    3
Babé Stick Labial 4 Gramos                               2
Gel Masaje Madagascar Sweetness Control 200ml            2
Nivea Crema 150ml                                        2
Gel Contorno de Ojos Sesderma Men 15ml                   2
Babaria Crema Facial Antiarrugas Rosa Mosqueta 50ml      2
Sebamed Champú Dermatológico Anticaspa                   2
Name: name, dtype: int64

In [65]:
products[products.duplicated(["name"])]

product_id                    sku  \
6159        35799              213000-x3   
6417         4868       Phytocolor-PH963   
7560         4490  sebamed.champu-372656   
8557        74810               175341x2   
9725        24277                 168460   
...           ...                    ...   
26078        3455                 179135   
26569       39913             178408Azu3   
26897       44255       MN12104U.Naranja   
26924       71104          CB23263X.niña   
27067        5012                 378406   

                                                    name    marca_value  \
6159   Epaplus Magnesio + Acido Hialuronico + Vitamin...       Epa-plus   
6417                               Phytocolor Kit Color           Phyto   
7560             Sebamed Champú Dermatológico Anticaspa         Sebamed   
8557   Singuladerm XpertSun Perfection Natural SPF 50 5g    Singuladerm   
9725               Crema Facial Malva Blanca Weleda 50ml         Weleda   
...                                                  ...            ...   
26078                     Farmatint 8N Rubio Claro 130ml      Farmatint   
26569  Suavinex Biberón +0m Tetina Redonda Silicona 3...       Suavinex   
26897  Platos Termosensibles White Hot Munchkin 2 Uni...       Munchkin   
26924  Chupete Silicona Little Cutie Canpol Babies 6-...  Canpol Babies   
27067        Sesderma Hidraloe Crema Contorno Ojos 15 ml       Sesderma   

                                       short_description    analytic_category  \
6159   <p><strong>Complemento alimenticio</strong> el...           Herbolario   
6417                <p>Tinte para el pelo de Phyto\n</p>                  NaN   
7560   <p>Champú sin jabón anticaspa para el lavado d...                  NaN   
8557   <p>Fotoprotector mineral en polvo ultraligero,...                  NaN   
9725   <p>Crema facial para bebés, sin perfume, que p...             Infantil   
...                                                  ...                  ...   
26078  <p>Farmatint es el gel de coloración permanene...  Cosmética y Belleza   
26569  <p>Biberón Suavinex 3 Posiciones, que permite ...             Infantil   
26897  <p>Platos de colores llamativos con banda indi...             Infantil   
26924  <p>Chupete con tetina anatómica de silicona in...                  NaN   
27067  <p>Tu mirada más rejuvenecida e iluminada grac...  Cosmética y Belleza   

                                                 picture  
6159   https://www.mifarma.es/media/catalog/product/2...  
6417   https://www.mifarma.es/media/catalog/product/p...  
7560   https://www.mifarma.es/media/catalog/product/s...  
8557   https://www.mifarma.es/media/catalog/product/1...  
9725   https://www.mifarma.es/media/catalog/product/1...  
...                                                  ...  
26078  https://www.mifarma.es/media/catalog/product/1...  
26569  https://www.mifarma.es/media/catalog/product/1...  
26897  https://www.mifarma.es/media/catalog/product/m...  
26924  https://www.mifarma.es/media/catalog/product/c...  
27067  https://www.mifarma.es/media/catalog/product/3...  

[90 rows x 7 columns]

### marca

In [74]:
products["marca_value"].value_counts().head(50)

Suavinex          761
Chicco            698
Sesderma          474
Isdin             461
Nuk               405
Apivita           372
La Roche Posay    364
Mam               350
Soria Natural     348
InterApothek      322
Nuxe              311
Beter             302
Arkopharma        287
Otras Marcas      287
Canpol Babies     285
Uriage            267
Ziaja             263
Vichy             259
Th Pharma         252
Tommee tippee     251
Avene             245
Avent             240
Eucerin           239
Bioderma          239
Pranarom          222
Sotya             222
Sensilis          206
Bell              203
Lacer             197
Solgar            192
Be+               189
Saro              187
Nûby              185
iap pharma        180
Rene Furterer     178
Klorane           178
Weleda            172
Caudalie          169
Rilastil          167
Babaria           167
Mustela           166
Lamberts          164
DrBrown´s         158
Lierac            155
Olmitos           154
Phyto     

In [75]:
products["marca_value"].isna().sum()

32

In [70]:
products[products["marca_value"].isna()]

product_id                            sku  \
622          9467                         162269   
923          7105                         157899   
1029         1031     SERV262253X2-262253-262253   
4376         7108                         152810   
4451          513         317171x2-317171-317171   
5063         8852                       1401LAT2   
5738        23899                          DUGIN   
5863        23893                          DUPAG   
5939        23884                        DUCAR01   
7308         8851                       1401LAT3   
9188        17205                          DUGAS   
9820        33677                 CONFETTI.M.CIR   
9992         8882                      125DEBLA4   
10281        5862                    KIT153827x2   
11338       17208                       DUDEPECO   
11680        7103                         152811   
12641       11609                          DUVIT   
13094       78484                       152937.0   
14110        9817  L648+L658+L668-L648-L658-L668   
16104       33681                 CONFETTI.L.CIR   
16276        5664                         160250   
16774       80525                          10010   
17680       80522                          10011   
18124        7106                  7611278903530   
19842       33683                CONFETTI.XL.CIR   
20447       33679                 CONFETTI.S.CIR   
20563       33681                 CONFETTI.L.CIR   
20926        1031     SERV262253X2-262253-262253   
21037       23902                       DUEQUECO   
24478        9467                         162269   
24622       41185                          DULIN   
26700       96969                         190406   

                                                    name marca_value  \
622                                Prelox 60 Comprimidos         NaN   
923                       Dr Dunner Cinnulin 40 Capsulas         NaN   
1029   Interapothek Champu Anticaida 400 ml + 400 ml ...         NaN   
4376                       Dr Dunner Menosoy 60 Capsulas         NaN   
4451   La Roche Posay Solucion Micelar 200ml + 200 ml...         NaN   
5063              Medela Bravado Design Seamless Latte M         NaN   
5738            Dr Dunner Ginkocel Biloba 40 Comprimidos         NaN   
5863                    Dr Dunner Pagosid 80 Comprimidos         NaN   
5939     Dr Dunner Silygold Cardo Mariano 80 Comprimidos         NaN   
7308              Medela Bravado Design Seamless Latte L         NaN   
9188                 Dr Dunner Gastrodiet 40 Comprimidos         NaN   
9820   Sujetador de Lactancia Medela Bravado Confetti...         NaN   
9992        Medela Bravado Design Bliss Negro 100 D/E DD         NaN   
10281  Medela Discos Absorbentes Desechables 30 Disco...         NaN   
11338                      Dr Dunner Derinol 40 Capsulas         NaN   
11680               Dr Dunner Nattolin Osteo 30 Capsulas         NaN   
12641  Dr Dunner Multivital Complemento Alimenticio 4...         NaN   
13094  Sublime Curl Champu Sublimador Rizos Rene Furt...         NaN   
14110  Lierac Magnificence Crema Aterciopelada 50ml +...         NaN   
16104  Sujetador de Lactancia Medela Bravado Confetti...         NaN   
16276                                   Flammaspray 75ml         NaN   
16774                  MiRebotica Gel Efecto Frío 250 ml         NaN   
17680                  MiRebotica Gel Efecto Calor 250ml         NaN   
18124                    Dr Dunner Neurodyn  30 Capsulas         NaN   
19842  Sujetador de Lactancia Medela Bravado Confetti...         NaN   
20447  Sujetador de Lactancia Medela Bravado Confetti...         NaN   
20563  Sutia de amamentacao Medela ameixa de confete ...         NaN   
20926      Champú Revitalizante Interapothek 400ml Duplo         NaN   
21037     Dr Dunner Echinol Bio Equinacea 80 Comprimidos         NaN   
24478                              Prelox-60 comprimidos         NaN   
24622                     Dr Dunner Linomega 60 Cápsulas  

### category

In [80]:
products["analytic_category"].value_counts()

Cosmética y Belleza    8336
Higiene                5410
Infantil               4901
Herbolario             2801
Nutrición              1600
Ortopedia               572
Vida Íntima             340
Óptica                  244
Perfumeria              165
Veterinaria              47
Name: analytic_category, dtype: int64

In [87]:
products["analytic_category"].unique()

array(['Infantil', nan, 'Cosmética y Belleza', 'Óptica', 'Higiene',
       'Herbolario', 'Ortopedia', 'Nutrición', 'Vida Íntima',
       'Veterinaria', 'Perfumeria'], dtype=object)

In [89]:
products["analytic_category"].isna().sum()

3537

In [92]:
products[products["analytic_category"].isna()].head()

product_id     sku                                               name  \
4        24517  011905                       Dodot Pañal T4 9-15Kg 30unds   
5        11938  011896         Dodot Activity T3 De 4 A 10 Kg 56 Unidades   
6        13109  011894              Dodot Pañal Sensitive T1 2-5Kg 30Unds   
14       12051  013687  Dodot Sensitive Pañal Primeras Semanas T/0 Has...   
31       13122  011907          Dodot Pañal Pack Semanal T/5 13-18kg 25ud   

   marca_value                                  short_description  \
4        Dodot  <p>¿Qué le pides a un pañal? ¿Absorción máxima...   
5        Dodot  <p>Pañales que mantienen la piel del culito se...   
6        Dodot  <p>Pañales para recién nacido que evitan la ir...   
14       Dodot  <p>Pañales Dodot para una máxima absorción de ...   
31       Dodot                      <p>Pañales pack semanal\n</p>   

   analytic_category                                            picture  
4                NaN  https://www.mifarma.es/media/catalog/product/0...  
5                NaN  https://www.mifarma.es/media/catalog/product/0...  
6                NaN  https://www.mifarma.es/media/catalog/product/0...  
14               NaN  https://www.mifarma.es/media/catalog/product/0...  
31               NaN  https://www.mifarma.es/media/catalog/product/0...

In [102]:
products[products["short_description"].str.find("<strong>") > 1].head(10)

product_id     sku                                               name  \
0         7349  165774     Pañales CHELINO LOVE Talla3 4-10 KG 36unidades   
1         7347  165776    Pañales CHELINO LOVE Talla5 13-18 KG 30unidades   
2        50282  230154                       Pañales Pingo Maxi T4 40 uds   
3         7348  165775     Pañales CHELINO LOVE Talla4 9-15 KG 34unidades   
7        60743  183816  Cumlaude Sunlaude Niños Ultrafluido Mineral SP...   
8        50297  230130                       Pañales Pingo Midi T3 44 uds   
11       33565  176748             Cumlaude Visilaude Gotas Oculares 10ml   
12       65365   14507  Desodorante de Avena Roll On Instituto Español...   
14       12051  013687  Dodot Sensitive Pañal Primeras Semanas T/0 Has...   
15       43862  180156       Dodot Sensitive Pañal Recien Nacido T2 34uds   

          marca_value                                  short_description  \
0             Chelino  <p>Los pañales <strong>Chelino</strong>, que p...   
1             Chelino  <p>Los pañales Chelino que presentan un novedo...   
2               Pingo  <p>Pingo es el primer pañal ecológico con cuat...   
3             Chelino  <p>Los pañales Chelino, que presentan un noved...   
7            Rilastil  <p>Muy alta protección solar formulada con <st...   
8               Pingo  <p>Pingo es el primer pañal ecológico con cuat...   
11           Rilastil  <p>Las gotas ocuares Visilaude de Cumlaude son...   
12  Instituto Español  <p>Desodorante a base de avena para la higiene...   
14              Dodot  <p>Pañales Dodot para una máxima absorción de ...   
15              Dodot  <p>Pañales ultra-absorbentes y suaves como la ...   

      analytic_category                                            picture  
0              Infantil  https://www.mifarma.es/media/catalog/product/1...  
1              Infantil  https://www.mifarma.es/media/catalog/product/1...  
2              Infantil  https://www.mifarma.es/media/catalog/product/2...  
3              Infantil  https://www.mifarma.es/media/catalog/product/1...  
7   Cosmética y Belleza  https://www.mifarma.es/media/catalog/product/1...  
8              Infantil  https://www.mifarma.es/media/catalog/product/2...  
11               Óptica  https://www.mifarma.es/media/catalog/product/1...  
12              Higiene  https://www.mifarma.es/media/catalog/product/1...  
14                  NaN  https://www.mifarma.es/media/catalog/product/0...  
15             Infantil  https://www.mifarma.es/media/catalog/product/1...

In [109]:
products[(products["short_description"].str.find("<strong>") > 1) & (products["analytic_category"].isna())]

product_id                sku  \
14          12051             013687   
62          13104             011899   
373          5429             002872   
393          8079             165889   
413          5424             002870   
...           ...                ...   
27575       95442               FRUS   
27589       13104             011899   
27636        7657           167925x2   
27740        3970  SERV348549-348549   
27769       49207              04028   

                                                    name  \
14     Dodot Sensitive Pañal Primeras Semanas T/0 Has...   
62                           Dodot Pañal T2 3-6Kg 40Unds   
373    Carbonato de magnesio 75 comprimidos Ana Maria...   
393    Gynea Serelys Perimenopausia Menopausia Sofoco...   
413    Magnesio Cloruro 147 Comprimidos Ana María La ...   
...                                                  ...   
27575                   Lubricante Frutopia Sandía 100ml   
27589                       Dodot fralda T2 3-6Kg 40Unds   
27636  Bioderma Atoderm creme com dispensador 500 ml ...   
27740  Tamanho do bolsão 15ml de Acne inflamatório Bi...   
27769    Soria Natural Natural extrato de echinacea 50ml   

                marca_value  \
14                    Dodot   
62                    Dodot   
373    Ana Maria LaJusticia   
393                 Serelys   
413    Ana Maria LaJusticia   
...                     ...   
27575                  Love   
27589                 Dodot   
27636              Bioderma   
27740              Bioderma   
27769         Soria Natural   

                                       short_description analytic_category  \
14     <p>Pañales Dodot para una máxima absorción de ...               NaN   
62     <p>Pañales Dodot que mantienen la piel seca du...               NaN   
373    <p>Complemento alimenticio de magnesio en form...               NaN   
393    <p>Suplemento alimenticio formulado con vitami...               NaN   
413    <p>Magnesio Cloruro Ana María La Justicia está...               NaN   
...                                                  ...               ...   
27575  <p>Lubricante íntimo con sabor a sandía para q...               NaN   
27589  <p>Pañales Dodot que mantienen la piel seca du...               NaN   
27636  <p><strong>Cuidado complementario</strong> par...               NaN   
27740  <p><strong>El primer tratamiento purificante <...               NaN   
27769  <p>Extracto natural a base de Equinácea que ay...               NaN   

                                                 picture  
14     https://www.mifarma.es/media/catalog/product/0...  
62     https://www.mifarma.es/media/catalog/product/0...  
373    https://www.mifarma.es/media/catalog/product/0...  
393    https://www.mifarma.es/media/catalog/product/1...  
413    https://www.mifarma.es/media/catalog/product/0...  
...                                                  ...  
27575  https://www.mifarma.es/media/catalog/product/f...  
27589  https://www.mifarma.es/media/catalog/product/0...  
27636  https://www.mifarma.es/media/catalog/product/1...  
27740  https://www.mifarma.es/media/catalog/product/s...  
27769  https://www.mifarma.es/media/catalog/product/0...  

[279 rows x 7 columns]

#### product_id - sku
Varios duplicados. El número es diferente al de sku duplicados.
#### name
Hay algunos missings que se pueden sustituir tomando marca y descripción (son pocos).

#### marca
Intentar sacar missing (son pocos) por name y description.
Visualización... Interesante ver cuánto se vende por cada marca y top productos más vendidos.

#### description
Importante... etiquetas como "<strong" nos pueden decir mucho para imputar variables faltantes...
No lo veo útil para visualización. Se podría imputar quitando etiquetas html (regex?), pero no lo veo...

#### category
Hay 3537 nulos... se podría tirar un modelo con palabras para imputar?

#### pict
Link de las img no sirve. Está desactualizado. img guardadas en otro sitio?
Ejemplo de una imagen de producto...https://d2csxpduxe849s.cloudfront.net/media/07EADC…ct%20High-79330F8C-1FDE-46EF-99CD8FACECA99C66.png

